cors.go

```diff
@@ -251,7 +255,17 @@ func AllowedOriginValidator(fn OriginValidator) CORSOption {
 	}
 }
 
-// ExposeHeaders can be used to specify headers that are available
+// OptionStatusCode sets a custom status code to use to respond to OPTIONS requests
+// by default it is set to 200 to ensure max compatibility, see:
+// https://stackoverflow.com/questions/46026409/what-are-proper-status-codes-for-cors-preflight-requests
+func OptionStatusCode(code int) CORSOption {
```

Comment:
I would rephrase this:

- When a handler returns and the status code has not been set prior to writing the body, Go automatically sets the status as HTTP 200. We should make it clear that you do not have to set this option unless you want non-default behaviour - i.e. HTTP 204.
- Refer to the RFC (in this case, the Fetch spec), not StackOverflow.

Reply:
Thanks for your feedback and your time !

For a bit more context, here is our use case:
![our use case](https://i.imgur.com/GQGVisi.png)

We need the CORS module to be able to set a status code on the OPTION requests so we do not add noise to our monitoring systems with 0 status codes all over the place. Indeed the client is served a 200 thanks to Go's default behaviour, but I believe this could be helpful to anyone that needs monitoring.

I will update the comment to reflect this better.

Reply:
OK, this makes more sense.

I would suggest that your monitoring middleware wraps the ResponseWriter and captures the final status code (and response duration too), rather than relying on what other middleware are doing.

Reply:
We already wrap the `ResponseWriter` in our monitoring middleware, but we can't use it in the case of an `OPTIONS` request since we do not perform any call on a backend (thus we never get a response).